Create Large Test Files.py

```diff
@@ -0,0 +1,255 @@
+		
+		
+		#1: convert letters to digits
+		#2: append '00' to right of string
+		#3:Apply the mathematical function mod=(n, 97) where n= the number obtained in step 2 above and 97 is the divisor.
+		#3a: Alternatively, to calculate without using the modulus operator, divide the numbers in step 2 above by 97.
+		#   Truncate the remainder to three digits and multiply it by .97. Round the result to the nearest whole number.
+		#4: Subtract the result in step 3 from 98. If the result is one digit, add a leading 0 to make it two digits.
+		#5: The two digits in the result from step 4 is the check digit. Append the resulting check digit to the
+		#   rightmost position in the combined string of characters described in step 1 above to generate the ULI.
+		
+		#digit_vals contains the conversion of numbers to letters
+		digit_vals = {
+		'A':10, 'H':17,'O':24,'V':31,'B':11,'I':18,'P':25,'W':32,'C':12,'J':19,'Q':26,'X':33,'D':13,'K':20,'R':27,'Y':34,
+		'E':14,'L':21,'S':28,'Z':35,'F':15,'M':22,'T':29,'G':16,'N':23,'U':30}
+		
+		uli_chars = list(ULI)
+		mod_uli_chars = []
+		for char in uli_chars:
+			if char.upper() in digit_vals.keys():
+				mod_uli_chars.append(str(digit_vals[char.upper()]))
+			else:
+				mod_uli_chars.append(char)
+		mod_uli_chars.append('00') 
+		digit_base = int("".join(mod_uli_chars))
+		digit_modulo = digit_base % 97
+		check_digit = 98 - digit_modulo
+
+		if valid:
+			return str(check_digit).zfill(2) #left pad check digit with 0 if length is less than 2
+		else:
+			return str(check_digit+6).zfill(2)[:2] #return a bad check digit (used in edit testing)
+
+#This function checks whether character strings in a column are unique to each other. 
+#When column strings are unique, the function returns a "Passed" statement.
+#If they are not, it returns a "Failed" statement. 
+def uli_check(df=None):
+    
+    """
+    This function passes in a single column dataframe to determine whether values within the column
+    are unique. 
+    
+    The function returns a "Passed" statement if they are, and returns a "Failed" statement if the values
+    are not unique. 
+    """
+    
+    df['unique'] = df
+    if len(df['unique'].unique()) == len(df.index): #Checks whether column values are unique. 
+        statement = "Unique Values: Passed"
+    else: 
+        statement = "Unique Values: Failed"
+    print(statement)
+    return df.drop('unique', axis = 1) #Drops the unique values column. 
+    print(df)
+
+
+# ### Creating Clean Test Files
+# 
+# The following defines a function that creates clean test files. The function uses 1000 row clean test files located in the repository for Bank 0 and Bank 1 to duplicate LAR data to produce data with a desired number of rows, and provide unique ULI's for each row. Test files may be created with rows as multiples of 1000. 
+
+# In[299]:
+
+
+def large_clean_test_files(bank=None, file=file, count=None):
+    
+    """
```

Comment:
Is it possible to add flexibility so that the user can enter any file size desired?

Reply:
We could - the function works to amplify an existing clean file. We can pass in a one row file to allow for any number of rows. The tradeoff would be that every row in the file would be the same except for the ULI.

Reply:
Can you handle the file length issue at the end? For example, a user wants 3500 records, you could create a 4000 record file and then remove the last 500 before saving.

Reply:
Yes! This works nicely. Thank you!